### ISSUE TITLE
Skip non-bucket dns entry in federated bucket list and gui

### ISSUE THREAD

**Issue by adiospeds** [minio/minio#7697]
We are using a federated setup of minio. I am serving our Minio GUI as well as s3 endpoint at project.qa.s3.domain.com 

We have setup a conditional forwarder in our bind dns such that all *.qa.s3.domain.com queries will be served by coredns. eg:

```
; Minio
qa.s3.domain.com.   NS ns1.coredns-qa.domain.com.
ns1.coredns-qa.domain.com.   IN  A   192.168.2.100
```

I have manually added an entry into etcd as I wanted s3 endpoint to be served from coredns server.

```
# On Etcd server:
export ETCDCTL_API=3

etcdctl put /skydns/com/domain/s3/qa/project/192.168.2.102 '{"host":"192.168.2.102","port":80,"ttl":30,"creationDate":"2019-05-24T07:48:23.35035718Z"}'

etcdctl get --keys-only --prefix /skydns
/skydns/com/domain/s3/qa/project/192.168.2.102

```

However added entry comes up as a blank bucket in the GUI as well as in the list buckets query to the endpoint
![image](https://user-images.githubusercontent.com/8983446/58421479-a7894700-80ad-11e9-9a46-2cb1abf1c85d.png)

```
╰─± aws s3 ls --endpoint-url=http://project.qa.s3.domain.com --no-verify-ssl --profile netops
2019-05-24 13:18:23

```

Please make it disappear from the `list buckets` call and `GUI`
This is what I had actually requested in https://github.com/minio/minio/pull/7339.

**Comment by kannappanr** [comment]
@adiospeds Thanks for reporting this issue, we will take a look at it.

**Comment by harshavardhana** [comment]
Was able to reproduce this issue @adiospeds

**Comment by adiospeds** [comment]
This patch fixed the issue for me .. Tested by upgrading to Version: 2019-06-01T03:46:14Z and now the blank bucket does not appear anymore 🥇

**Comment by adiospeds** [comment]
Hold, looks like this fix created a new issue.
I was able to create buckets after this patch, however I was not able to delete those buckets> when I try to delete, it gives me "No entries found for this key" error

more on this from slack discussion :

> 
> Aditya Pednekar   [Today at 8:28 PM]
> `No entries found for this key`
> 
> 18 replies
> 
> Anis Elleuch   [27 minutes ago]
> are you using mc to delete ? @apednekar
> 
> Aditya Pednekar   [27 minutes ago]
> here's what the logs say
> ```Jun  3 14:53:20 prod-sc-minio-netops-netflow-server-01 minio[18793]: API: SYSTEM()
> Jun  3 14:53:20 prod-sc-minio-netops-netflow-server-01 minio[18793]: Time: 14:53:20 UTC 06/03/2019
> Jun  3 14:53:20 prod-sc-minio-netops-netflow-server-01 minio[18793]: Error: No entries found for this key
> Jun  3 14:53:20 prod-sc-minio-netops-netflow-server-01 minio[18793]:        3: cmd/web-handlers.go:1976:cmd.toWebAPIError()
> Jun  3 14:53:20 prod-sc-minio-netops-netflow-server-01 minio[18793]:        2: cmd/web-handlers.go:1856:cmd.toJSONError()
> Jun  3 14:53:20 prod-sc-minio-netops-netflow-server-01 minio[18793]:        1: cmd/web-handlers.go:273:cmd.(*webAPIHandlers).DeleteBucket()```
> 
> Aditya Pednekar   [26 minutes ago]
> no I used the minio gui to delete it
> 
> Aditya Pednekar   [25 minutes ago]
> will DM you more details as the requests contain internal org domain names
> 
> Aditya Pednekar   [18 minutes ago]
> ```─○ aws s3 mb s3://demo02 --endpoint-url=http://proj.prod-sc.s3.domain.com --no-verify-ssl --profile proj
> make_bucket: demo02```
> had created the bucket using aws cli and echo $? was 0
> 
> Aditya Pednekar   [16 minutes ago]
> ```ls -l /mnt/data-01/
> total 0
> drwxr-xr-x 2 minio minio  6 Jun  3 14:53 demo02
> drwxr-xr-x 4 minio minio 32 May 31 22:54 netflow```
> I also see that the folder for the bucket is created on all minio servers
> 
> Anis Elleuch   [15 minutes ago]
> it looks like you are using etcd and for some reasons it is returning an error (edited)
> 
> Aditya Pednekar   [15 minutes ago]
> ```etcdctl get --keys-only --prefix /skydns
> /skydns/com/domain/s3/prod-sc/proj/10.11.12.13
> 
> /skydns/com/domain/s3/prod-sc/proj/demo02/10.11.12.13
> 
> /skydns/com/domain/s3/prod-sc/proj/netflow/10.11.12.13```
> (edited)
> 
> Aditya Pednekar   [14 minutes ago]
> etcd entry is created inside all etcd servers
> 
> Aditya Pednekar   [13 minutes ago]
> and coredns server is able to resolve the bucket
> ```dig +short @localhost demo02.proj.prod-sc.s3.domain.com
> 10.11.12.13```
> 
> Anis Elleuch   [11 minutes ago]
> I will try this locally right now and update you @apednekar
> 
> Aditya Pednekar   [11 minutes ago]
> cool .. Ichking etcd logs in the mean time - will keep you posted
> 
> Aditya Pednekar   [8 minutes ago]
> just a wild guess, issue could be bcas of this patch that had fixed my another issue
> 
> Aditya Pednekar   [8 minutes ago]
> https://github.com/minio/minio/pull/7699
> GitHub
> Skip non-bucket DNS entry by harshavardhana · Pull Request #7699 · minio/minio
> Description Skip non-bucket DNS entry Motivation and Context Fixes #7697 Regression Yes perhaps in 4c04896 How Has This Been Tested? Using a local etcd setup and bucket population Types of ...
> 
> Aditya Pednekar   [6 minutes ago]
> probably hitting this code when trying to delete buckets
>     ```if srvRecord.Key == "" || srvRecord.Key == etcdPathSeparator {
>             continue
>         }```
> (edited)
> 
> Aditya Pednekar   [2 minutes ago]
> after downgrading to minio version Version: 2019-04-18T21:44:59Z, i was able to delete the bucket
> 
> Aditya Pednekar   [1 minute ago]
> this was out previous minio version being used in prod
> 
> Aditya Pednekar   [1 minute ago]
> this means there was no issue on the etcd or coredns servers - its the new minio code

**Comment by adiospeds** [comment]
The fix probably created more issues. More on it in : https://github.com/minio/minio/issues/7741
